amazon-speech-api/amazonTranscribeHarris.js

```diff
@@ -2,29 +2,112 @@ const request = require('request')
-  };
-  // NOTE: You can only call this function once.  Afterwards, the VocabularyName is already taken
-  transcribeservice.createVocabulary(params, function(err, data) {
-    if (err) console.log(err, err.stack); // an error occurred
-    else     console.log(data);           // successful response
-  });
+let vocabularyObject = {
+  LanguageCode: 'en-US',
+  Phrases: ["candidates", "talkhiring","jobseekers","resume"], // Array of Strings
+  VocabularyName: generateRandomString(),
+}
+
+// Create a bucket for the audio file
+function createBucket(bucketName, audioFile){
+  s3.createBucket({Bucket: bucketName}, function(err, data) {
+  if (err) {
+     console.log("===We got err", err);
+     } else {
+       params = {Bucket: bucketName, Key: audioFile, Body: 'Hello!'};
+       s3.putObject(params, function(err, data) {
+           if (err) {
+               console.log(err)
+           } else {
+               console.log("Successfully uploaded bucket:", bucketName);
+           }
+        });
+     }
+  })
 }
 
-function startTranscriptionJob(transcriptionObject) {
+function getVocabulary(name, callback){
+	console.log(`===Checking if vocabulary ${name} is ready...`);
+	transcribeservice.getVocabulary({VocabularyName:name}, callback);
+}
+
+function deleteVocabulary(all=false, obj){
+	if (all){
+		deleteAll()
+	} else {
+		deleteFunction(obj);
+	}
+	function deleteFunction(obj){
+		transcribeservice.deleteVocabulary({VocabularyName:obj}, (err,data)=>{
+		if (err) console.log(err, err.stack); // an error occurred
+		else     console.log(data);           // successful response
+		});
+	}
+	
+	function deleteAll(){
+		transcribeservice.listVocabularies({},(err,data)=>{
+			if (err) {console.log(err, err.stack)} // an error occurred
+	  		else {
+	  			data.Vocabularies.forEach((vocabulary)=>{
+	  				transcribeservice.deleteVocabulary({
+	  					VocabularyName: vocabulary.VocabularyName
+	  				},(err,data)=>{
+	  				if (err) {console.log(err, err.stack)}
+	  				else {console.log(vocabulary, "Successfully Deleted")}
+	  				});
+	  			});
+	  		}
+		});
+	}
+}
+
+function listVocabularies(){
+	transcribeservice.listVocabularies({},(err,data)=>{
+			if (err) {console.log(err, err.stack)} // an error occurred
+	  		else {
+	  			console.log(data.Vocabularies)
+	  		}
+		})
+}
+// Create an request to start a transcription job to api
+function createVocabulary(params={}, callback){
+    transcribeservice.createVocabulary({
+      LanguageCode: params.LanguageCode,
+      Phrases: params.Phrases,
+      VocabularyName: params.VocabularyName
+      }, createVocabularyHandler);
+  }
+function createVocabularyHandler(err,data){
+    if (err){
+    	console.log(err, err.stack);
+    	if (err.message ==  'You have too many vocabularies. Delete a vocabulary and try your request again.'){
+    		//Auto Deleting Vocabularies
+    		console.log('===Auto-deleting Vocabularies to make space');
+    		deleteVocabulary(all=true)
```

Comment:
this seems a little dangerous.  just because we don't have room doesn't mean that we want to delete all the vocabularies

Reply:
Sorry, I meant to add that it was for development purposes, when testing the vocabularies can fill up quickly. The function does not delete all vocabularies I renamed it. It was intended initially to delete all vocabulary, however listVocabularies() responds with 5 vocabularies at a time. When it deleted 5, it was enough to continue developing w/o errors. I could adjust the function to delete all the vocabularies if you would like?  I renamed the function that does the deleting to deleteFive.

The function I intended for us to use in production, was the deleteVocabulary.deleteOne (renamed deleteFunction -> deleteOne). This function deletes one specific vocabulary from list. I also just added a deleteList() which will delete the desired list